### PR TITLE
add ParametrizedEvolution to mixed device

### PR DESCRIPTION
**Pull request by Qottmann**
`default.mixed` does not automatically support operations that provide matrices, we missed that during development.

### CODE REVIEW

File: tests/pulse/test_parametrized_evolution.py [hunk 1]
```diff
@@ -358,3 +358,37 @@ def circuit2(params):
...
+
+        @qml.qnode(default, interface="jax")
+        def qnode_def(x):
+            qml.pulse.ParametrizedEvolution(H_pulse, x, 5.0)
+            return qml.expval(qml.PauliZ(0))
+
+        @qml.qnode(mixed, interface="jax")
+        def qnode_mix(x):
+            qml.pulse.ParametrizedEvolution(H_pulse, x, 5.0)
+            return qml.expval(qml.PauliZ(0))
```
Comment (dwierichs) [inline tests/pulse/test_parametrized_evolution.py:384]:
```suggestion
        def circuit(x):
            qml.pulse.ParametrizedEvolution(H_pulse, x, 5.0)
            return qml.expval(qml.PauliZ(0))

        qnode_def = qml.QNode(circuit, default, interface="jax")
        qnode_mix = qml.QNode(circuit, mixed, interface="jax")
```
in order to facilitate reusing this and making it more stable to modifications of the circuit being used for both circuit :)

**Review by dwierichs — COMMENTED** [review]
Nice, thanks @Qottmann !
I just have a tiny suggestion for the test, and want to ask whether we want to add a test to `tests/devices/test_default_mixed_jax.py`, although it does feel a bit redundant...

File: tests/pulse/test_parametrized_evolution.py (hunk 1 above)
Comment (Qottmann) [reply tests/pulse/test_parametrized_evolution.py:384]: nice thanks!

**Review by lillian542 — APPROVED** [review]
Looks good to me once tests run! 🙃

**Pull request merged**